### PR TITLE
Create new libraries from existing code

### PR DESCRIPTION
**Pull request by tarsius**
`package-build.el` used to contain this comment:

> In future we should provide a hook, and perform this step in a separate package.

If done the second part and also did so for the recipe mode.

### CODE REVIEW

File: package-batches.el [hunk 1]
```diff
@@ -0,0 +1,57 @@
+;;; package-batches.el --- Create batches for packages
```
Comment (purcell) [inline package-batches.el]: "batches" should be "badges", but anyway I think these things should be placed in a MELPA-specific file, e.g. `melpa.el`.

**Review by purcell — CHANGES_REQUESTED** [review]

File: package-batches.el (hunk 1 above)
Comment (tarsius) [reply package-batches.el]: Made the same in the descriptions of Kickstarter rewards. Of course there I was not allowed to change it and now I have to figure out how to send batches to users :wink:

Comment (tarsius) [reply package-batches.el]: So I should just rename the file (but leave it in this repository)and reword some things a bit?

Comment (purcell) [reply package-batches.el]: > now I have to figure out how to send batches to users

For some of the rewards, you were offering batches of badges, tbh.

**Comment by purcell** [comment]
> So I should just rename the file (but leave it in this repository)and reword some things a bit?

Yes. I think it's okay to make it `package-build-badges` or similar: eventually it'd disappear from `package-build` into MELPA's own (future) hooked functionality.

**Comment by purcell** [comment]
Is it possible this has caused https://github.com/melpa/melpa/issues/5150?

**Comment by tarsius** [comment]
Must be the case. I guess Melpa loads `package-build` directly instead of relying on in it being on the `load-path`. I'll have a look and push a fix if I can figure out what's going on.

**Comment by tarsius** [comment]
Fixed with https://github.com/melpa/melpa/commit/c173db53ac32b6dc7dbc72131f12a87d817afe8f.

**Pull request merged**